src/mtype.d

```diff
@@ -4542,103 +4542,23 @@ extern (C++) class TypeArray : TypeNext
         }
         if (ident == Id.reverse && (n.ty == Tchar || n.ty == Twchar))
```

Comment:
In fact, these first two branches can be removed, as the error is the same for both strings and arrays in this PR.